packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -513,11 +690,10 @@ export function splitMoveDest<T>(
 export function replaceMoveDest<T>(table: MoveEffectTable<T>, id: MoveId, mark: Attach<T>): void {
     const origId = table.splitIdToOrigId.get(id);
     if (origId !== undefined) {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const effect = table.dstEffects.get(origId)!;
-        const index = effect.findIndex((p) => p.id === id);
-        assert(effect[index].replaceWith === undefined, "Move dest already replaced");
-        effect[index].replaceWith = [mark];
+        const effect = getOrAddEmptyToMap(table.dstEffects, origId);
+        const partition = getOrAddMovePartition(effect, id);
```

Comment:
I replaced all these because they started failing and I couldn't figure out why. Normally I would have reached out to you instead of just replacing it but this was during the winter break.
Right now, when I remove the usages of `getOrAddMovePartition` and `getOrAddEmptyToMap`, I get crashes in the `A ↷ [B, B⁻¹] === A` and `(A ↷ B) ↷ [B⁻¹, B] === A ↷ B` tests. The crashes happen in `updateMoveDestPairing`. It used to crash in more places than that so it's likely that some of the issues I fixed along the way (like that `??` precedence issue or the cases in compose where `replaceMoveSrc` was called without `composeMark`) were the underlying cause or they happened on invalid test cases that I hadn't yet filtered out.
I'll look into the `updateMoveDestPairing` crash.

Reply:
It seems that for those rebasing tests `updateMoveSrcPairing` adds an entry in `srcEffects` that then gets removed by `applyMoveEffectsToMark` during a `queue.pop()`. Is that not meant to happen? Is this an unintended consequence of not doing the move effects in a later pass?

Reply:
No, it gets applied when it gets removed.

Reply:
But my "fix" indeed hides the bug in functions where the index of the partition is needed.